hpedockerplugin/cmd/cmd_deleteshare.py

```diff
@@ -19,154 +22,179 @@ def __init__(self, file_mgr, share_info):
         self._fpg_name = share_info['fpg']
 
     def execute(self):
-        LOG.info("Delting share %s..." % self._share_info['name'])
+        share_name = self._share_info['name']
+        LOG.info("Deleting share %s..." % share_name)
         # Most likely nothing got created at the backend when share is
         # not in AVAILABLE state
-        if self._share_info['status'] != 'AVAILABLE':
-            self._delete_share_from_etcd(self._share_info['name'])
-            return json.dumps({u"Err": ''})
+        if self._share_info['status'] == 'FAILED':
+            LOG.info("Share %s is not in FAILED state. Removing from ETCD..."
+                     % share_name)
+            ret_val, status = self._delete_share_from_etcd(share_name)
+            return ret_val
+
+        elif self._share_info['status'] == 'CREATING':
+            msg = ("Share %s is in CREATING state. Please wait for it to be "
+                   "in AVAILABLE or FAILED state and then attempt remove."
+                   % share_name)
+            LOG.info(msg)
+            return json.dumps({"Err": msg})
 
-        with self._fp_etcd.get_fpg_lock(
-                self._backend, self._cpg_name, self._fpg_name):
-            self._remove_quota()
+        try:
             self._delete_share()
+        except exception.ShareBackendException as ex:
+            return json.dumps({"Err": ex.msg})
+        except Exception as e:
+            msg = 'Failed to remove share %(share_name)s from backend: %(e)s' \
+                  % ({'share_name': share_name, 'e': six.text_type(e)})
+            LOG.error(msg)
+
```

Comment:
@imran-ansari here if share deletion is not successful and raises other than ShareBackendException should not we return error response?

Reply:
Actually it will never be any exception other than ShareBackendException. General exception block itself is not required - will be removing it.